zk_toolbox/crates/common/src/forge.rs

```diff
@@ -93,6 +98,39 @@ impl ForgeScript {
         });
         self
     }
+    // Do not start the script if balance is not enough
+    pub fn private_key(&self) -> Option<H256> {
+        self.args.args.iter().find_map(|a| {
+            if let ForgeScriptArg::PrivateKey { private_key } = a {
+                Some(H256::from_str(private_key).unwrap())
+            } else {
+                None
+            }
+        })
+    }
+
+    pub fn rpc_url(&self) -> Option<String> {
+        self.args.args.iter().find_map(|a| {
+            if let ForgeScriptArg::RpcUrl { url } = a {
+                Some(url.clone())
+            } else {
+                None
+            }
+        })
+    }
+
+    pub async fn check_the_balance(&self, minimum_value: U256) -> anyhow::Result<(bool, Address)> {
+        let Some(rpc_url) = self.rpc_url() else {
+            return Ok((true, Address::zero()));
+        };
+        let Some(private_key) = self.private_key() else {
+            return Ok((true, Address::zero()));
+        };
```

Comment:
In this case, shouldn't we be returning `false` if parameters are not correctly set?

Reply:
Could this ever happen that there's no rpc_url or private_key set?

Reply:
no idea...probably if we want to have it without broadcasting